lib/pub_sub/event_class_factory.rb

```diff
@@ -17,6 +17,10 @@ def initialize(event_name, domain_name:, abstract_event_class:)
     end
 
     def build_event_class
+      event_class = res_event_class_name.safe_constantize
+
+      return event_class unless event_class.nil?
+
       event_class = event_class_name.safe_constantize
 
       return event_class unless event_class.nil?
```

Comment:
Here is a similar case as above. Of course it's up to you.
```suggestion
      return event_class if event_class.present?
```